### PR TITLE
Filesystem Trim Support

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#836

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM, merge this until the engine dependency update

File: generate_grpc.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ if [ ! -e ${DIR} ]; then
     rm -rf ${DIR}
     mkdir -p $DIR
     wget https://raw.githubusercontent.com/longhorn/longhorn-engine/master/proto/ptypes/common.proto -P $DIR
-    wget https://raw.githubusercontent.com/longhorn/longhorn-engine/master/proto/ptypes/controller.proto -P $DIR
+    wget https://raw.githubusercontent.com/shuo-wu/longhorn-engine/trim-dev/proto/ptypes/controller.proto -P $DIR
```
Comment (innobead) [inline generate_grpc.sh:17]: This needs to change.

**Review by innobead — CHANGES_REQUESTED** [review]

File: generate_grpc.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@ if [ ! -e ${DIR} ]; then
     rm -rf ${DIR}
     mkdir -p $DIR
     wget https://raw.githubusercontent.com/longhorn/longhorn-engine/master/proto/ptypes/common.proto -P $DIR
-    wget https://raw.githubusercontent.com/longhorn/longhorn-engine/master/proto/ptypes/controller.proto -P $DIR
+    wget https://raw.githubusercontent.com/longhorn/longhorn-engine/trim-dev/proto/ptypes/controller.proto -P $DIR
```
Comment (innobead) [inline generate_grpc.sh:17]: This is not found `https://raw.githubusercontent.com/longhorn/longhorn-engine/trim-dev/proto/ptypes/controller.proto`, do we need to change this? I think it should be kept at master

**Review by innobead — CHANGES_REQUESTED** [review]

File: generate_grpc.sh (hunk 2 above)
Comment (shuo-wu) [reply generate_grpc.sh:17]: My fault... Should be `master`.

**Review by innobead — APPROVED** [review]

**Pull request merged**